### ISSUE TITLE
Correction issues with Cucumber/Calabash

### ISSUE THREAD

**Issue by Purus** [bbatsov/rubocop#2824]
We have a Calabash project which was formatted using Eclipse DLTK Ruby plugin editor.

Rubocop works and auto-correct many of the issues but gives problem for Cucumber step definitions.

```
drop0-Security.rb:8:6: W: (...) interpreted as grouped expression.
Given (/^The URL for application is available$/)do
     ^
drop0-Security.rb:8:7: C: Don't use parentheses around a literal.
Given (/^The URL for application is available$/)do
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Is there any specific options to overcome these and any particular configs available for Calabash/Cucumber steps?

**Comment by alexdowad** [comment]
Is there a reason you write `Given (regex) do` rather than `Given(regex) do` or `Given regex do`?

**Comment by Purus** [comment]
We use Eclipse for the development and the files were auto-formatted with the Ruby and Cucumber plugins in Eclipse.

**Comment by alexdowad** [comment]
Looks like RC doesn't agree with the style of the Eclipse plugins' generated code.

**Comment by alexdowad** [comment]
By the way, RC can also auto-format your code.

**Comment by Purus** [comment]
Yeah, I was about to update my post. The automatic corrections are not
perfect. It makes the calabash test unusable with syntax error .

**Comment by alexdowad** [comment]
If the auto-correction creates a syntax error, that is a bug in RC and should be reported. We will fix it.

**Comment by Purus** [comment]
Sorry that I could not respond with the error details now. I am replying
now in my mobile device and will add more details about the issue tomorrow
from my work.

**Comment by alexdowad** [comment]
Can you show us the syntax error which is created by RC's autocorrection?

**Comment by Purus** [comment]
The line below

`Then (/^I Login into the Application$/) do`

is auto-corrected as below.

`Then /^I Login into the Application$/ do`

**Comment by alexdowad** [comment]
That's not a syntax error, unless there is more than what you are showing us.

```
22:25 ~prog/Ruby/rubocop % pry
[1] pry(main)> def Then(regex)
[1] pry(main)*   # silly implementation, just for testing
[1] pry(main)*   if "abc" =~ regex
[1] pry(main)*     yield
[1] pry(main)*   end  
[1] pry(main)* end  
=> :Then
[2] pry(main)> Then /a/ do
[2] pry(main)*   puts 'block called'
[2] pry(main)* end  
block called
=> nil
[3] pry(main)> Then (/a/) do
[3] pry(main)*   puts 'block called again'
[3] pry(main)* end  
block called again
=> nil
```

**Comment by Purus** [comment]
Please let me know what additional information you need.

The auto-corrected files are shown as syntax-error when running the
calabash tests.

Thanks,
R.Purusothaman

**Comment by alexdowad** [comment]
@Purus Please upload the entire autocorrected file somewhere and post a link here.

**Comment by jonas054** [comment]
I'm going to guess what the problem was. I think that before #2814 was fixed, if you auto-corrected
```ruby
Given (/^The URL for application is available$/)do
end
```
with `Style/SpaceAroundKeyword` disabled, you'd get
```ruby
Given /^The URL for application is available$/do
end
```
which is invalid ruby. But since `Style/RedundantParentheses` doesn't do this auto-correction anymore, I'm closing the issue.
